electrumx/server/session.py

```diff
@@ -213,29 +216,158 @@ def _ssl_context(self):
-                                                    service.port, ssl=sslc)
-            except OSError as e:    # don't suppress CancelledError
-                self.logger.error(f'{kind} server failed to listen on {service.address}: {e}')
+            if service.protocol == 'http':
+                host = None if service.host == 'all_interfaces' else str(service.host)
+                try:
+                    app = web.Application(middlewares=[
+                        error_middleware(self),
+                        request_middleware(self)
+                    ])
+                    handler = HttpHandler(self, self.db, self.mempool, self.peer_mgr, kind)
+                    # GET
+                    app.router.add_get('/proxy', handler.proxy)
+                    app.router.add_get('/proxy/health', handler.health)
+                    app.router.add_get('/proxy/blockchain.block.header', handler.block_header)
+                    app.router.add_get('/proxy/blockchain.block.headers', handler.block_headers)
+                    app.router.add_get('/proxy/blockchain.estimatefee', handler.estimatefee)
+                    # app.router.add_get('/proxy/headers.subscribe', handler.headers_subscribe)
+                    # app.router.add_get('/proxy/relayfee', handler.relayfee)
+                    app.router.add_get('/proxy/blockchain.scripthash.get_balance', handler.scripthash_get_balance)
+                    app.router.add_get('/proxy/blockchain.scripthash.get_history', handler.scripthash_get_history)
+                    app.router.add_get('/proxy/blockchain.scripthash.get_mempool', handler.scripthash_get_mempool)
+                    app.router.add_get('/proxy/blockchain.scripthash.listunspent', handler.scripthash_listunspent)
+                    app.router.add_get('/proxy/blockchain.scripthash.subscribe', handler.scripthash_subscribe)
+                    app.router.add_get('/proxy/blockchain.transaction.broadcast', handler.transaction_broadcast)
+                    app.router.add_get('/proxy/blockchain.transaction.get', handler.transaction_get)
+                    app.router.add_get('/proxy/blockchain.transaction.get_merkle', handler.transaction_merkle)
+                    app.router.add_get('/proxy/blockchain.transaction.id_from_pos', handler.transaction_id_from_pos)
+                    # app.router.add_get('/proxy/server.add_peer', handler.add_peer)
+                    app.router.add_get('/proxy/server.banner', handler.banner)
+                    app.router.add_get('/proxy/server.donation_address', handler.donation_address)
+                    app.router.add_get('/proxy/server.features', handler.server_features_async)
+                    app.router.add_get('/proxy/server.peers.subscribe', handler.peers_subscribe)
+                    app.router.add_get('/proxy/server.ping', handler.ping)
+                    # app.router.add_get('/proxy/server.version', handler.server_version)
```

Comment:
But server version would be nice to expose?

Reply:
The server version interface will be directed to the common interface. The common interface forwards requests to the TCP interface and returns the server response from the TCP version. Of course, we can also implement the code to return the server version directly in the HTTP server section, which would be even better.